zerver/decorator.py

```diff
@@ -903,8 +915,15 @@ def _wrapped_view_func(
 # secret, and also the originating IP (for now).
 @has_request_variables
 def authenticate_notify(request: HttpRequest, secret: str = REQ("secret")) -> bool:
-    return is_local_addr(request.META["REMOTE_ADDR"]) and constant_time_compare(
-        secret, settings.SHARED_SECRET
+    # The secret either will be raw SHARED_SECRET or it will be a HMAC computed using SHARED_SECRET as both key and data
+    shared_secret: str = settings.SHARED_SECRET
+    shared_secret_bytes = shared_secret.encode("utf-8")
+    computed_hmac = hmac.new(
+        shared_secret_bytes, shared_secret_bytes, digestmod="sha256"
+    ).hexdigest()
+    return is_local_addr(request.META["REMOTE_ADDR"]) and (
+        constant_time_compare(secret, settings.SHARED_SECRET)
+        or constant_time_compare(secret, computed_hmac)
```

Comment:
Can you explain the design around adding this `hmac` mechanism here, rather than using the `SHARED_SECRET` mechanism again? 

If the `tusd` server only supports that as a mechanism, I think it might makes sense to generate a new secret to be shared between `tusd` and `django`, rather than reusing the Tornado one -- and having each secret only work for the endpoint(s) where it's needed.

That sort of isolation of secrets would be a lot safer for thinking about the risks of one process or the other being compromised.